include/ceed/jit-source/magma/magma-basis-interp-nontensor.h

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and other CEED contributors.
+// All Rights Reserved. See the top-level LICENSE and NOTICE files for details.
+//
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This file is part of CEED:  http://github.com/ceed
+
+/// @file
+/// Internal header for MAGMA non-tensor basis interpolation
+#ifndef _ceed_magma_basis_interp_nontensor_h
+#define _ceed_magma_basis_interp_nontensor_h
+
+#include "magma-common-nontensor.h"
+
+////////////////////////////////////////////////////////////////////////////////
+extern "C" __launch_bounds__(MAGMA_BASIS_BOUNDS(BASIS_Q, MAGMA_MAXTHREADS_1D)) __global__
+    void magma_interp_nontensor_n(int n, CeedScalar const* dA, int ldda, CeedScalar const* dB, int lddb, CeedScalar* dC, int lddc) {
+  MAGMA_DEVICE_SHARED(CeedScalar, shared_data);
+
+  const int tx      = threadIdx.x;
+  const int ty      = threadIdx.y;
```

Comment:
I find it confusing that the `interp` and `deriv` kernels are both in a file named `magma-basis-interp-nontensor.h`.  But perhaps you were already planning to change that?

At the moment, it looks like `interp` and `deriv` are identical except for the fact that they need to be compiled with different values for `QCOMP` and `NB`, no?  Were you thinking to refactor into a common templated function like you did for the `ref` backends, then have `interp` and `deriv` just instantiate the kernel with different values?

I know I said previously that I thought it would be good to have "scalar" and "vector" versions (i.e., a kernel where `QCOMP` is 1 and a kernel where it is > 1) so that the "scalar" one could avoid the `QCOMP` loop altogether.  That also allows us to use `=` instead of `+=` in the transpose mode application for the scalar kernel, when we know we only have one component.  Of course I don't know how much difference that would really make in the end.  Hopefully very little, if the compiler is smart...  

Also, I realize looking through the kernels again that that wouldn't affect the MAGMA code, anyway, so I guess there's no benefit at all to not rearranging into a common core function like you did for `ref`.  (And *if* there is any penalty in `ref`, well...it's `ref`.  So probably not a big deal.)

Reply:
I'll attempt to respond in order:
1. Yeah I can rename that file. It seems like a better name would be `magma-basis-interp-deriv-nontensor.h`, right?
2. That's right `interp` and `deriv` are identical aside from the compile time constants. I can play around with combining them to reduce the code duplication.
3. I agree the performance difference of the loop with size = 1 shouldn't change much, since it's known at compile time. For the `+=` vs. `=` in both the `ref` and `magma` cases, let me see what I can do it would be nice to fix this.